apps/zotonic_core/src/support/z_media_preview.erl

```diff
@@ -387,13 +387,19 @@ out_mime(Mime, Options, Context) ->
 %% @doc Return the preferred mime type of the image generated by resizing an image of a certain type and size.
 -spec out_mime( InMime :: binary(), list() ) -> { OutMime :: binary(), Extension :: string()}.
 out_mime(Mime, Options) ->
-    out_mime1(get_lossless_value(Options), Mime).
+    Mime1 = case proplists:get_value(webp, Options, false) of
```

Comment:
I wonder if we should make this a "format" option to enable more formats in the future?

Reply:
Ah.. yes... that would be more future proof indeed.

Reply:
We can map an extension to mime format, using the `[Mime|_] = mimetypes:ext_to_mimes(<<"webp">>)` or just use a list of supported formats of which we know it works, with a fallback to webp or jpeg.

I think the list of known formats might be best.

Reply:
Right now the "lossless" option decides if the extension will be png or jpg depending on the setting. Right now I added webp to this in a backward compatible way. Modern image formats have the possibility to choose between lossy and lossless compression. We might want to change this configuration setting for `Z 1.1`.